src/Symfony/Installer/DownloadCommand.php

```diff
@@ -545,6 +545,107 @@ protected function getUrlContents($url)
     }
 
     /**
+     * It returns the project's Composer config as a PHP array.
+     *
+     * @return $this|array
+     */
+    protected function getProjectComposerConfig()
+    {
+        $composerJsonFilepath = $this->projectDir.'/composer.json';
+
+        if (!is_writable($composerJsonFilepath)) {
+            if ($this->output->isVerbose()) {
+                $this->output->writeln(sprintf(
+                    " <comment>[WARNING]</comment> Project's Composer config cannot be updated because\n".
+                    " the <comment>%s</comment> file is not writable.\n",
+                    $composerJsonFilepath
+                ));
+            }
+
+            return $this;
+        }
+
+        return json_decode(file_get_contents($composerJsonFilepath), true);
+    }
+
+    /**
+     * It saves the given PHP array as the project's Composer config. In addition
+     * to JSON-serializing the contents, it synchronizes the composer.lock file to
+     * avoid out-of-sync Composer errors.
+     *
+     * @param array $config
+     */
+    protected function saveProjectComposerConfig(array $config)
+    {
+        $composerJsonFilepath = $this->projectDir.'/composer.json';
+        $this->fs->dumpFile($composerJsonFilepath, json_encode($config, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES)."\n");
```

Comment:
These JSON constants are available starting with PHP 5.4, so 5.3 users will get an error.
 
Maybe use the int flag manually : `128 | 64` and add a comment about the constants?